tests/core/test_expectation_suite.py

```diff
@@ -26,7 +33,290 @@ def exp1():
 
 
 @pytest.fixture
-def exp2():
+def expect_column_values_to_be_in_set_col_a_with_meta_dict() -> dict:
+    # Note, value_set is distinct to ensure this is treated as a different expectation
+    return {
+        "expectation_type": "expect_column_values_to_be_in_set",
+        "kwargs": {
+            "column": "a",
+            "value_set": [1, 2, 3, 4, 5],
+            "result_format": "BASIC",
+        },
+        "meta": {"notes": "This is an expectation."},
+    }
+
+
+class TestInitMethod:
+    """Tests related to ExpectationSuite.__init__()"""
+
+    @pytest.mark.unit
+    def test_expectation_suite_init_defaults(self, fake_expectation_suite_name: str):
+        suite = ExpectationSuite(expectation_suite_name=fake_expectation_suite_name)
+
+        default_meta = {"great_expectations_version": ge_version}
+
+        assert suite.expectation_suite_name == fake_expectation_suite_name
+        assert suite._data_context is None
+        assert suite.expectations == []
+        assert suite.evaluation_parameters == {}
+        assert suite.data_asset_type is None
+        assert suite.execution_engine_type is None
+        assert suite.meta == default_meta
+        assert suite.ge_cloud_id is None
+
+    @pytest.mark.unit
+    def test_expectation_suite_init_overrides(
+        self,
+        fake_expectation_suite_name: str,
+        expect_column_values_to_be_in_set_col_a_with_meta: ExpectationConfiguration,
+    ):
+        class DummyDataContext:
+            pass
+
+        class DummyExecutionEngine:
+            pass
```

Comment:
Should these inherit the classes they're meant to model? Is there value in using `mock.MagicMock` instead?

Reply:
For these dummy classes (just meant to fill the param list) we don't want to inherit from the class with all the overhead / need to implement abstract methods. For this test we don't need to call any methods on them, just to check that what we supply in the param is then stored as an attribute on the class instance.
As for `mock.MagicMock`, that can definitely be used instead here but we don't need the extra functionality of setting return values or checking what methods were called / params passed. So a dummy class might be less overhead. Maybe @billdirks has an opinion!